enhancements/console/customize-project-roles.md

```diff
@@ -0,0 +1,160 @@
+---
+title: customize-project-roles
+authors:
+  - "@jerolimov"
+reviewers:
+  - "@spadgett"
+  - "@cvogt"
+approvers:
+  - "@spadgett"
+  - "@cvogt"
+creation-date: 2021-02-25
+last-updated: 2021-02-25
+status: implementable
+---
+
+# Customize Project Access Roles
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The OpenShift Dev Console provides a simplified way to share access the selected project. Users can be assigned to specific roles under the Project menu item on the Project access tab. Currently only the roles "Admin", "Edit" and "View" are selectable.
+
+Cluster admins want give project admins only access to a customized list of predefined roles to force compliance rules or provide more options.
+
+## Motivation
+
+Customers who uses their own kubernetes roles wants that the project admin can use this more specific options within the Developer Console.
+
+### Goals
+
+- Add configuration options to allow cluster admins to customize the project access roles.
+- Provide a list of default values to the admin
+
+### Non-Goals
+
+- Introducing a UI to configure this project access roles without a yaml editor.
+
+## Proposal
+
+### Extending the Console CRD
+
+Similar to [Customize Developer Catalog Categories](./catalog-categories.md) we extend the existing `operator.openshift.io/v1` / `Console` CRD. It provides already a `spec.customization` area where we can add a list of roles.
+
+The list of roles should link to `rbac.authorization.k8s.io/v1` / `Role` or `ClusterRole`.
+
+So the final yaml should look like this one:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    brand: online
+    projectAccess:
```

Comment:
I'd give this even more context like

```yaml
customization:
  developerPerspective:
    clusterRoleOptions:
      - admin
      - edit
      - view
```

Someone better at names might come up something better :) But I think it would be good to make it clear this only applies to the dev perspective project access form, not the general create role binding form.

Reply:
Combined both, until we have a better naming:

```yaml
    developerPerspective:
      projectAccessRoles:
```

Or do you prefer `clusterRoleOptions`? My thought here is that its really restrictive to `ClusterRoles`. Which is maybe also fine when we don't want like `Roles` :thinking:

Reply:
What to you think about `projectAdmin.availableClusterRoles` ? It's a more functional description then using the perspective?